.github/workflows/add_label_precommit.yml

```
@@ -0,0 +1,38 @@
+name: Add label on pre-commit success
+on:
+    workflow_run:
+        workflows: [pre-commit]
+        types: [requested, completed]
+jobs:
+    add-label-on-pre-commit-success:
+        runs-on: ubuntu-latest
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        steps:
+            -   name: Add label
+                uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+                with:
+                    script: |
+                        github.rest.issues.addLabels({
+                            owner: context.repo.owner,
+                            repo: context.repo.repo,
+                            issue_number: context.issue.number,
+                            labels: ['pre-commit-passed']
+                        })
+                env:
+                    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    remove-label-on-pre-commit-failure:
```

Since this runs on the `requested` event, I believe this will remove the label whenever the job is pending and not just failed, right? If so, that seems right by design, though the job name could be tweaked to something like
```suggestion
    remove-label-on-pre-commit-start:
```

---

The workflow is supposed to be triggered if `pre-commit` starts or finishes (for some reason it's not triggering right now...). So this job should run either if pre-commit `completed` and wasn't a `success` or if pre-commit was `requested`. 

I'm not sure what name we could pick that encompasses both of those situations.

---

I have renamed it `remove-label-on-pre-commit-not-success` to cover both scenarios